packages/itwinui-css/src/button/base.scss

```diff
@@ -41,30 +32,40 @@
   svg {
     @include mixins.iui-icon-style('m');
     transition: fill var(--iui-duration-1) ease-out;
-    fill: var(--_iui-button-icon-fill, currentColor);
+    fill: var(--_iui-button-icon-fill, var(--iui-color-icon));
+
+    button:hover &,
+    button:active & {
```

Comment:
This whole block looks a bit hacky, and it doesn't work for advanced cases like `<Button as="a">`. Maybe it should be handled in `iui-field` instead?

Reply:
Did this 👆🏻